addons/purchase_stock/models/account_invoice.py

```diff
@@ -151,3 +151,16 @@ def _get_last_step_stock_moves(self):
         for invoice in self.filtered(lambda x: x.type == 'in_refund'):
             rslt += invoice.mapped('invoice_line_ids.purchase_line_id.move_ids').filtered(lambda x: x.state == 'done' and x.location_dest_id.usage == 'supplier')
         return rslt
+
+
+class AccountInvoiceLine(models.Model):
+    _inherit = "account.invoice.line"
+
+    def get_invoice_line_account(self, type, product, fpos, company):
+        purchase_line = self._context.get('purchase_line_id')
+        owner = purchase_line and self.env['purchase.order.line'].browse(purchase_line).move_ids.mapped('move_line_ids.owner_id')
```

Comment:
What if some lines have an owner, and some don't? Or Some move lines are processed and others are not (because of a backorder)? I think the latter could be addressed (filter the moves based on their state), but the first one...